src/v/kafka/server/handlers/configs/config_response_utils.cc

```diff
@@ -312,6 +312,48 @@ override_if_not_default(const std::optional<T>& override, const T& def) {
     }
 }
 
+// The cases in which the override is considered equivalent to the default are:
+// 1. `tristate` is empty (`std::nullopt`) and the optional default is also
+// empty (`std::nullopt`)
+// 2. `tristate` is disabled (`disable_tristate`) and the optional default is
+// empty (`std::nullopt`).
+// 3. `tristate` holds a value and the optional default holds the same value.
+// clang-format off
+// Refer to the following truth table for a comprehensive explanation of the result of this function;
+// or rather, how the property source will appear to a user querying the topic configuration.
+// override       def            result
+// {disabled}     {value}        DYNAMIC_TOPIC_CONFIG
+// {empty}        {value}        DYNAMIC_TOPIC_CONFIG
+// {disabled}     {nullopt}      DEFAULT_CONFIG
+// {empty}        {nullopt}      DEFAULT_CONFIG
+// {value}        {value}        DEFAULT_CONFIG
```

Comment:
This seems a little surprising, though I suspect this is the case because it's how it's done elsewhere? I would've thought that any explicit override (which I'm assuming is coming straight from the user?) would be listed as a dynamic config. Curious if you found that odd too, and if you have any historical links to that decision if so

Reply:
Double checking this.

Reply:
I will have to double check whether this is exhibited in the `kafka` equivalent property though.

Reply:
Thanks. It's very possible this is correct UX as is, but just want to make sure